src/libraries/Microsoft.PowerFx.Core/Public/Types/UserProvidedTypeDefinitions/FormulaTypeToSchemaHelper.cs

```diff
@@ -0,0 +1,153 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.PowerFx.Core.Binding;
+using Microsoft.PowerFx.Core.Public.Types;
+using Microsoft.PowerFx.Core.Utils;
+using Microsoft.PowerFx.Types;
+
+namespace Microsoft.PowerFx.Core
+{
+    internal static class FormulaTypeToSchemaHelper
+    {
+        public static FormulaTypeSchema ToSchema(this FormulaType type, DefinedTypeSymbolTable definedTypeSymbols)
+        {
+            return ToSchema(type, definedTypeSymbols, maxDepth: 5);
+        }
+                
+        public static FormulaType ToFormulaType(this FormulaTypeSchema schema, DefinedTypeSymbolTable definedTypeSymbols)
+        {
+            var typeName = schema.Type.Name;
+
+            if (TryLookupType(typeName, definedTypeSymbols, out var actualType))
+            {
+                if (schema.Type.IsTable)
+                {
+                    return actualType switch
+                    {
+                        RecordType recordType => recordType.ToTable(),
```

Comment:
>RecordType recordType => recordType.ToTable(), [](http://example.com/codeflow?start=24&length=46)

Is actual type here the member type?
For example, for *[A:n] type, the actualType is n